pkg/controller/staticpod/staticpod_controller.go

```diff
@@ -0,0 +1,593 @@
+		Spec: corev1.PodSpec{
+			HostPID:            true,
+			HostNetwork:        true,
+			RestartPolicy:      corev1.RestartPolicyNever,
+			ServiceAccountName: UpgradeServiceAccount,
+			Containers: []corev1.Container{{
+				Name:    UpgradeWorkerContainerName,
+				Command: []string{"/bin/sh", "-c"},
+				VolumeMounts: []corev1.VolumeMount{
+					{
+						Name:      hostPathVolumeName,
+						MountPath: hostPathVolumeMountPath,
+					},
+					{
+						Name:      configMapVolumeName,
+						MountPath: configMapVolumeMountPath,
+					},
+				},
+				ImagePullPolicy: corev1.PullIfNotPresent,
+				SecurityContext: &corev1.SecurityContext{
+					Privileged: &True,
+				},
+				Image: UpgradeWorkerImage,
+			}},
+			Volumes: []corev1.Volume{{
+				Name: hostPathVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: hostPathVolumeSourcePath,
+					},
+				}},
+			},
+		},
+	}
+)
+
+func Format(format string, args ...interface{}) string {
+	s := fmt.Sprintf(format, args...)
+	return fmt.Sprintf("%s: %s", controllerName, s)
+}
+
+// Add creates a new StaticPod Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	if !utildiscovery.DiscoverGVK(controllerKind) {
+		return nil
+	}
+	return add(mgr, newReconciler(mgr))
+}
+
+var _ reconcile.Reconciler = &ReconcileStaticPod{}
+
+// ReconcileStaticPod reconciles a StaticPod object
+type ReconcileStaticPod struct {
+	client.Client
+	scheme   *runtime.Scheme
+	recorder record.EventRecorder
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileStaticPod{
+		Client:   utilclient.NewClientFromManager(mgr, controllerName),
+		scheme:   mgr.GetScheme(),
+		recorder: mgr.GetEventRecorderFor(controllerName),
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r, MaxConcurrentReconciles: concurrentReconciles})
+	if err != nil {
+		return err
+	}
+
+	// 1. Watch for changes to StaticPod
+	if err := c.Watch(&source.Kind{Type: &appsv1alpha1.StaticPod{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return err
+	}
+
+	// 2. Watch for changes to node
+	// When node turn ready, reconcile all StaticPod instances
+	// nodeReadyPredicate filter events which are node turn ready
+	nodeReadyPredicate := predicate.Funcs{
+		UpdateFunc: func(evt event.UpdateEvent) bool {
+			return nodeTurnReady(evt)
+		},
+	}
+
+	reconcileAllStaticPods := func(c client.Client) []reconcile.Request {
+		staticPodList := &appsv1alpha1.StaticPodList{}
+		err := c.List(context.TODO(), staticPodList)
+		if err != nil {
+			return nil
+		}
```

Comment:
```go
if err := c.List(context.TODO(), staticPodList); err != nil {
}
```
I think the upper format will reduces the scope of error

Reply:
Maybe we need to do error return here, instead of just returning nil?

Reply:
It's hard to deal with this error here and I will try to figure it out